### ISSUE TITLE
Provision HostPort service only once pods are Ready

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#18850]
As discussed in https://github.com/cilium/cilium/issues/16191, we should provision the HostPort service for pods only once they are in Ready state.

We currently provision it as soon as the pods are created. Because of that, if a HostPort service also exists for an InitContainer in the pod, then the HostPort service of the Container will point to the InitContainer's. That might be unexpected for users.

That can be observed with the following manifest:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: myapp-pod
  labels:
    app: myapp
spec:
  containers:
  - name: myapp-container
    image: docker.io/cilium/json-mock:1.2
    ports:
    - containerPort: 80
      hostPort: 8080
  initContainers:
  - name: init-myservice
    image: docker.io/cilium/json-mock:1.2
    ports:
    - containerPort: 80
      hostPort: 8888
```
The pod remains indefinitely in state Init:0/1, exposing the container's port 80 on port 8888 of the host. However, since we configure the datapath service mapping for the container and the init container at the same time, the service ends up being also exposed on port 8080.
```
$ ks exec cilium-whhp2 -- curl 10.164.0.5:8888 > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2109  100  2109    0     0  2059k      0 --:--:-- --:--:-- --:--:-- 2059k
$ ks exec cilium-whhp2 -- curl 10.164.0.5:8080 > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2109  100  2109    0     0  1029k      0 --:--:-- --:--:-- --:--:-- 2059k
```

This issue is fairly low priority since it's probably a corner case to have HostPorts on InitContainer and even rarer to hit this incorrect provisioning then.
